### PR TITLE
Mmccam vs minimal

### PR DESCRIPTION
**Pull request by igalic**
- switch amd64's GENERIC-MMCCAM to GENERIC
- restore the old GENERIC-MMCCAM as MINIMAL-MMCCAM
- remove CAMDEBUG_FLAGS, while keeping CAMDEBUG itself.

### CODE REVIEW

**Comment by igalic** [comment]
a previous attempt at this was made as https://reviews.freebsd.org/D40909

**Comment by bsdimp** [comment]
This looks good to my eye. We need to land mmc cam as the default for 15 thought.

**Comment by igalic** [comment]
> We need to land mmc cam as the default for 15 though

I am not going to submit a "GENERIC-MMC-CLASSIC" patch until branch, and the current PR needs to be merged for 14

(and maybe MFC'd)

File: sys/amd64/conf/GENERIC-MMCCAM [hunk 1]
```diff
@@ -1,34 +1,24 @@
-# MMCCAM is the kernel config for doing MMC on CAM development
-# and testing on bhyve
+#
+# GEMERIC-MMCCAM
```
Comment (jlduran) [inline sys/amd64/conf/GENERIC-MMCCAM:2]: s/GEMERIC/GENERIC/

Comment (igalic) [reply sys/amd64/conf/GENERIC-MMCCAM:2]: should fix it here https://github.com/freebsd/freebsd-src/blob/5d0826017fc9b84f50ef19b0d452bc0590072f2d/sys/arm/conf/GENERIC-MMCCAM#L2 as well then

Comment (igalic) [reply sys/amd64/conf/GENERIC-MMCCAM]: done.

**Comment by igalic** [comment]
@bsdimp ping

**Comment by bsdimp** [comment]
I like this, but have been kinda swamped. Will try to land it in next pull request pass.

**Comment by igalic** [comment]
@bsdimp , now that 14 is in ALPHA, we should get this merged before we branch proper.

**Comment by igalic** [comment]
> > We need to land mmc cam as the default for 15 though
> 
> I am not going to submit a "GENERIC-MMC-CLASSIC" patch until we branch, and the current PR needs to be merged for 14
> 
> (and maybe MFC'd)

@bsdimp now that 14 is branched and this still isn't merged, I might as well do what you suggested.

**Comment by igalic** [comment]
i think this will probably need a RELNOTES entry.

File: sys/amd64/conf/GENERIC-MMCCAM [hunk 2]
```diff
@@ -1,33 +1,23 @@
...
-# CAM-specific stuff
-device		pass
-device		scbus
-device		da
-
 options 	MMCCAM
 
 # Add CAMDEBUG stuff
 options 	CAMDEBUG
 options 	CAM_DEBUG_FLAGS=(CAM_DEBUG_INFO|CAM_DEBUG_PROBE|CAM_DEBUG_PERIPH)
```
Comment (bsdimp) [inline sys/amd64/conf/GENERIC-MMCCAM:17]: I'd drop the last of these lines. CAMDEBUG is fine.. it make it easy to turn on debugging. CAM_DEBUG_FLAGS make things too chatty by default. Yes, the one that's there should likely have that done to it too.

File: sys/amd64/conf/GENERIC-MMCCAM [hunk 3]
```diff
@@ -1,33 +1,23 @@
...
+
+# pass(4) device
+device		pass
```
Comment (bsdimp) [inline sys/amd64/conf/GENERIC-MMCCAM:20]: Is pass not in generic?

File: sys/amd64/conf/MINIMAL [hunk 4]
```diff
@@ -36,6 +36,7 @@ makeoptions	WITH_CTF=1		# Run ctfconvert(1) for DTrace support
 options 	SCHED_ULE		# ULE scheduler
 options 	NUMA			# Non-Uniform Memory Architecture support
 options 	PREEMPTION		# Enable kernel thread preemption
+options 	VIMAGE			# Subsystem virtualization, e.g. VNET
```
Comment (bsdimp) [inline sys/amd64/conf/MINIMAL:39]: this seems unrelated.

File: sys/amd64/conf/MINIMAL-MMCCAM [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# MMCCAM is the kernel config for doing MMC on CAM development
```
Comment (bsdimp) [inline sys/amd64/conf/MINIMAL-MMCCAM:1]: I don't think this is needed. Can you remove it form your submission?

File: sys/i386/conf/MINIMAL [hunk 6]
```diff
@@ -41,6 +41,7 @@ makeoptions	WITH_CTF=1		# Run ctfconvert(1) for DTrace support
 
 options 	SCHED_ULE		# ULE scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
+options 	VIMAGE			# Subsystem virtualization, e.g. VNET
```
Comment (bsdimp) [inline sys/i386/conf/MINIMAL:44]: again, unrelated change.

File: sys/amd64/conf/MINIMAL (hunk 4 above)
Comment (igalic) [reply sys/amd64/conf/MINIMAL:39]: that's why it's a standalone commit

**Comment by bsdimp** [comment]
left some comments. And I'm on the fence if you should turn off CAM_DEBUG_FLAGS with this commit or as a separate commit.

File: sys/i386/conf/MINIMAL (hunk 6 above)
Comment (igalic) [reply sys/i386/conf/MINIMAL:44]: do you want me to create a separate pull request for that?

**Comment by igalic** [comment]
thank you for the feedback
I'll try to make something sensible of it tomorrow

File: sys/amd64/conf/GENERIC-MMCCAM (hunk 2 above)
Comment (igalic) [reply sys/amd64/conf/GENERIC-MMCCAM]: removed, in its own commit, so you can decide whether that's a good idea or not as you review this PR

File: sys/amd64/conf/GENERIC-MMCCAM (hunk 3 above)
Comment (igalic) [reply sys/amd64/conf/GENERIC-MMCCAM]: it is, but it's not in MINIMAL, which is what amd64's GENERIC-MMCCAM kernel was based on.

File: sys/amd64/conf/MINIMAL (hunk 4 above)
Comment (igalic) [reply sys/amd64/conf/MINIMAL]: spun out into #962

File: sys/i386/conf/MINIMAL (hunk 6 above)
Comment (igalic) [reply sys/i386/conf/MINIMAL]: spun out into #962

**Comment by bsdimp** [comment]
I think this is close from my quick glance now. I'll take a more detailed look. Sorry this has lingered.

**Comment by igalic** [comment]
don't worry, I've lost all feeling for time about 5 years ago. I didn't notice this has been open for 8 months, until somebody pointed it out

**Comment by bsdimp** [comment]
OK. Now that I've had time to look at this with a clear head it's mostly good. And I'll land the good parts.
I'm going to drop the MINIMAL-MMCCAM config, though, since I don't think we have a huge need to preserve the old behavior. We can revisit that detail later in the name of getting the rest in easily enough. 834517eff8aa.
